mlrun/artifacts/model.py

```diff
@@ -612,6 +614,12 @@ def _get_extra(target, extra_data, is_dir=False):
     return extra_dataitems
 
 
+def _remove_tag_from_spec_yaml(model_spec):
+    spec_dict = model_spec.to_dict()
+    spec_dict.pop("tag", None)
+    return yaml.dump(spec_dict)
```

Comment:
to_dict has an `exclude` parameter.
I suggest you add the parameter to `to_yaml` and exclude the tag in line 332 instead of doing the conversion multiple times